LEGO1/lego/legoomni/src/entity/legocarraceactor.cpp

```diff
@@ -27,20 +31,68 @@ LegoCarRaceActor::LegoCarRaceActor()
 	VariableTable()->SetVariable(g_fuel, "0.8");
 }
 
-// STUB: LEGO1 0x10080590
-void LegoCarRaceActor::FUN_10080590(float)
+// FUNCTION: LEGO1 0x10080590
+// FUNCTION: BETA10 0x100cd8cf
+void LegoCarRaceActor::FUN_10080590(float p_float)
 {
+	MxFloat maxSpeed = m_maxLinearVel;
+	Mx3DPointFloat destEdgeUnknownVector = Mx3DPointFloat();
+	Mx3DPointFloat worldDirection = Mx3DPointFloat(m_roi->GetWorldDirection());
```

Comment:
This should be written as `Mx3DPointFloat worldDirection(m_roi->GetWorldDirection());`

Reply:
Oddly enough, this is not equivalent - your suggestion produces a slightly worse match. Please let me know if you still want to change it. The mysteries of MSVC 4.2 ...

Reply:
You are right, I guess the compiler doesn't optimize this code to be equivalent the way it should be in modern compilers. We'll leave it as-is then!